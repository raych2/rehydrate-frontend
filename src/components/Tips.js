const Tips = () => {
  return (
    <>
      <div className="topicSection smallerTopicSection">
        <div className="mainHeader">
          How to Save <div className="rectangle">Water</div>?
        </div>
        <div className="infoContainer">
          <div className="subHeader">Take a quiz</div>
          <div className="subText">
            Test your knowledge on our website and record your score!
          </div>
        </div>
        <div className="infoContainer">
          <div className="subHeader">Read</div>
          <div className="subText">
            Rehydrate holds infographics updated with our research on saving
            water and related topics.
          </div>
        </div>
        <div className="infoContainer">
          <div className="subHeader">Share</div>
          <div className="subText">
            Our infographics and your quiz score are shareable to social media.
            Teach your peers what they can do to "Rehydrate!"
          </div>
        </div>
      </div>
    </>
  );
};

export default Tips;
