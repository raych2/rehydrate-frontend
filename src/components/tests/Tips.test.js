import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Router, MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Tips from '../Tips';

describe('Tips component', () => {
  test('renders correctly', () => {
    const { container } = render(<Tips />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
  test('navigates to Quiz page after link is clicked', async () => {
    const history = createMemoryHistory();
    history.push = jest.fn();
    render(
      <Router location={history.location} navigator={history}>
        <Tips />
      </Router>
    );
    const user = userEvent.setup();
    await user.click(
      screen.getByRole('link', {
        name: /take a quiz/i,
      })
    );
    expect(history.push).toHaveBeenCalledWith(
      {
        hash: '',
        pathname: '/quiz',
        search: '',
      },
      undefined
    );
  });
});
