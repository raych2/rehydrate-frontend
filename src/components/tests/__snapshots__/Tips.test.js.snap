// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tips component renders correctly 1`] = `
<div>
  <div
    class="topicSection smallerTopicSection"
  >
    <div
      class="mainHeader"
    >
      How to Save 
      <div
        class="rectangle shorterTitle"
      >
        Water
      </div>
      ?
    </div>
    <div
      class="infoContainer"
    >
      <a
        class="subHeader quizLink"
        href="/quiz"
      >
        Take a quiz
      </a>
      <div
        class="subText"
      >
        Test your knowledge on our website and record your score!
      </div>
    </div>
    <div
      class="infoContainer"
    >
      <div
        class="subHeader"
      >
        Read
      </div>
      <div
        class="subText"
      >
        Rehydrate holds infographics updated with our research on saving water and related topics.
      </div>
    </div>
    <div
      class="infoContainer"
    >
      <div
        class="subHeader"
      >
        Share
      </div>
      <div
        class="subText"
      >
        Our infographics and your quiz score are shareable to social media. Teach your peers what they can do to "Rehydrate!"
      </div>
    </div>
  </div>
</div>
`;
